concourse/pipelines/create-bosh-cloudfoundry.yml

```diff
@@ -0,0 +1,901 @@
+    source:
+      bucket: {{state_bucket}}
+      region_name: {{aws_region}}
+      versioned_file: cf-manifest.yml
+
+  - name: cf-release
+    type: git
+    source:
+      uri: https://github.com/cloudfoundry/cf-release
+      tag_filter: {{cf-release-version}}
+
+jobs:
+  - name: init
+    serial_groups: [bosh-deploy]
+    serial: true
+    plan:
+      - get: paas-cf
+        trigger: {{auto_deploy}}
+      - task: bootstrap-s3-state
+        config:
+          platform: linux
+          image: docker:///governmentpaas/curl-ssl
+          inputs:
+            - name: paas-cf
+          run:
+            path: sh
+            args:
+              - -e
+              - -c
+              - |
+                paas-cf/concourse/scripts/s3init.sh {{state_bucket}} bosh.tfstate paas-cf/concourse/init_files/terraform.tfstate.tpl
+                paas-cf/concourse/scripts/s3init.sh {{state_bucket}} bosh-manifest-state.json paas-cf/concourse/init_files/bosh-init-state.json.tpl
+                paas-cf/concourse/scripts/s3init.sh {{state_bucket}} bosh-secrets.yml paas-cf/concourse/init_files/zero_bytes
+                paas-cf/concourse/scripts/s3init.sh {{state_bucket}} bosh-CA.tar.gz paas-cf/concourse/init_files/empty.tar.gz
+                paas-cf/concourse/scripts/s3init.sh {{state_bucket}} cf-secrets.yml paas-cf/concourse/init_files/zero_bytes
+                paas-cf/concourse/scripts/s3init.sh {{state_bucket}} cf.tfstate paas-cf/concourse/init_files/terraform.tfstate.tpl
+                paas-cf/concourse/scripts/s3init.sh {{state_bucket}} cf-certs.tar.gz paas-cf/concourse/init_files/empty.tar.gz
+      - put: pipeline-trigger
+        params: {bump: patch}
+
+  - name: bosh-terraform
+    serial_groups: [bosh-deploy]
+    serial: true
+    plan:
+      - aggregate:
+          - get: pipeline-trigger
+            trigger: true
+            passed: [ init ]
+          - get: paas-cf
+            passed: [init]
+          - get: vpc-tfstate
+          - get: concourse-tfstate
+          - get: bosh-tfstate
+
+      - task: extract-terraform-variables
+        config:
+          platform: linux
+          image: docker:///ruby#2.2-slim
+          inputs:
+            - name: paas-cf
+            - name: vpc-tfstate
+            - name: concourse-tfstate
+          outputs:
+            - name: terraform-variables
+          run:
+            path: sh
+            args:
+              - -e
+              - -c
+              - |
+                ruby paas-cf/concourse/scripts/extract_tf_vars_from_terraform_state.rb \
+                < vpc-tfstate/vpc.tfstate \
+                > terraform-variables/vpc.tfvars.sh
+                ruby paas-cf/concourse/scripts/extract_tf_vars_from_terraform_state.rb \
+                < concourse-tfstate/concourse.tfstate \
+                > terraform-variables/concourse.tfvars.sh
+      - task: terraform-apply
+        config:
+          platform: linux
+          image: docker:///governmentpaas/terraform
```

Comment:
We have a spike [#113285957](https://www.pivotaltracker.com/story/show/113285957) to investigate it